pandas/tests/internals/test_external_block.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=W0102
+
+import numpy as np
+
+import pandas as pd
+from pandas.core.internals import Block, BlockManager
+
+
+class CustomBlock(Block):
+
+    def formatting_values(self):
+        return np.array(["Val: {}".format(i) for i in self.values])
+
+
+def test_custom_repr():
+    values = np.arange(3, dtype='int64')
+
+    # series
+    block = CustomBlock(values, placement=slice(0, 3))
```

Comment:
In general it has proven a bit difficult to construct Series and DataFrame objects from given blocks, without re-creating the blocks (eg in Series, the block gets converted to array, which is then passed to SingleBlockManager, which does not preserve the block type)